File: ui_ux/lib/models/notification.dart
class NotificationModel {
  final String senderId;
  final String recipientId;
  final String messageType;
  final String message;

  NotificationModel({
    required this.senderId,
    required this.recipientId,
    required this.messageType,
    required this.message,
  });

  factory NotificationModel.fromJson(Map<String, dynamic> json) {
    return NotificationModel(
      senderId: json['senderId'],
      recipientId: json['recipientId'],
      messageType: json['messageType'],
      message: json['message'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'senderId': senderId,
      'recipientId': recipientId,
      'messageType': messageType,
      'message': message,
    };
  }
}

File: ui_ux/backend/models/Notification.js
const mongoose = require("mongoose");

const Schema = mongoose.Schema;

const notificationSchema = new Schema({
  senderId: {
    type: String,
    required: true,
  },
  recipientId: {
    type: String,
    required: true,
  },
  messageType: {
    type: String,
    enum: [
      "teacherAcceptedOffer",
      "teacherRejectedOffer",
      "studentAcceptedApplication",
      "teacherAppliedAdvertisement",
      "studentSentOffer",
    ],
    required: true,
  },
  message: {
    type: String,
    required: true,
  },
});
module.exports = mongoose.model("Notification", notificationSchema);
